src/cursor/cur_join.c

```diff
@@ -1333,7 +1333,7 @@ __wt_curjoin_open(WT_SESSION_IMPL *session,
 	WT_ERR(__wt_scr_alloc(session, 0, &tmp));
 	if (columns != NULL) {
 		WT_ERR(__wt_struct_reformat(session, table,
-		    columns, strlen(columns), NULL, 1, tmp));
+		    columns, strlen(columns), NULL, false, tmp));
```

Comment:
@agorrod, projections only include columns specified, so keys are not included unless mentioned.  The parameter flag indicates if keys are allowed.  `__wt_struct_reformat` also processes column lists for column groups, which is why that flag is needed.